app/controllers/ansible_roles_controller.rb

```diff
@@ -22,18 +22,71 @@ def destroy
 
   def import
     changed = @importer.import!
-    if changed.values.all?(&:empty?)
+    rows = prepare_rows(changed)
+    if rows.empty?
       success no_changed_roles_message
       redirect_to ansible_roles_path
     else
-      render :locals => { :changed => changed }
+      render :locals => { :rows => rows }
     end
   end
 
   def confirm_import
     @importer.finish_import(params[:changed]&.to_unsafe_h)
-    success _('Import of roles successfully finished.')
-    redirect_to ansible_roles_path
+    @variables_importer.import_variables_roles(params[:changed]) if params[:changed]['new'] || params[:changed]['old']
+    success _('Import of roles and their variables was successfully finished.')
+  end
+
+
+  def get_variable_action(kind)
+    { 'new' => _('Add'), 'obsolete' => _('Remove'), 'update' => _('Update') }[kind]
+  end
+
+  def get_old_roles_variables(roles, role)
+    variables = { 'Add' => [], 'Remove' => [], 'Update' => [] }
+    imported_variables = @variables_importer.import_variable_names(roles)
+    imported_variables.each do |kind, temp_variables|
+      temp_variables.each do |temp_variable|
+        variables[get_variable_action(kind)].append(temp_variable.key) if temp_variable.ansible_role_id == role.id
+      end
+    end
+  end
+
+  def get_string_variables(variables)
+    str = ''
+    variables.each do |action, temp_variables|
+      str += "#{action}: #{temp_variables.size} " unless temp_variables.empty?
+    end
+  end
+
+  def get_role_action(kind)
+    { 'new' => _('Import Role '), 'obsolete' => _('Remove Role'), 'old' => _('Update Role Variables') }[kind]
+  end
+
+  def prepare_rows(changed)
+    rows = []
+    changed.each do |kind, roles|
+      roles.each do |role|
+        role_action = get_role_action(kind)
+        if kind == 'new'
+          variables = { 'Add' => @importer.get_variables_names(role)[role.name] }
+        elsif kind == 'obsolete'
+          variables = { 'Remove' => role.ansible_variables.map(&:key) }
+        elsif kind == 'old'
+          variables = get_old_roles_variables(roles, role)
```

Comment:
The proxy API is called for each existing role with a request for variable names of all existing roles, but all variables that do not belong to `role` role are discarded. It should be enough to make just one call to get everything that is needed, can you think of a way?